### PR TITLE
feat(utxo-lib): add RPC tests

### PR DESCRIPTION
**Pull request by OttoAllmendinger**
Add test suite for typical 2-of-3 multisig variants and validate against
regtest node.

Capture RPC responses as test fixtures.

Issue: BG-34148

### CODE REVIEW

**Review by tylerlevine — DISMISSED** [review]
wow, this is great. Very nice, this is going to be extremely useful

:1st_place_medal:

File: modules/utxo-lib/test/integration_local_rpc/generate/main.ts [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+
+  if (diff > 0) {
+    console.log(`mining ${diff} blocks to reach height ${minBlocks}`);
+    const address = await rpc.getNewAddress();
+    await rpc.generateToAddress(diff, address);
+  }
+}
+
+function toRegtestAddress(network: Network, scriptType: ScriptType, script: Buffer): string {
+  if (scriptType === 'p2wsh' || scriptType === 'p2wkh') {
```
[the remark this answers is not in the document]
Comment (OttoAllmendinger) [reply modules/utxo-lib/test/integration_local_rpc/generate/main.ts:40]: let's not forget about `bech32m` ;)

File: modules/utxo-lib/test/integration_local_rpc/generate/RpcClient.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+import * as assert from 'assert';
+
+import axios, { AxiosError } from 'axios';
+
+const utxolib = require('../../../src');
+const coins = require('../../../src/coins');
+
+import { Network } from './types';
+
+function debug(...args: unknown[]) {
```
[the remark this answers is not in the document]
Comment (OttoAllmendinger) [reply modules/utxo-lib/test/integration_local_rpc/generate/RpcClient.ts:13]: sure why not

**Review by tylerlevine — APPROVED** [review]

File: modules/utxo-lib/package.json [hunk 3]
```diff
@@ -52,16 +52,18 @@
     "wif": "^2.0.1"
   },
   "devDependencies": {
+    "axios": "^0.21.1",
```
Comment (tylerlevine) [inline modules/utxo-lib/package.json:55]: should add `debug` here

File: modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.ts [hunk 4]
```diff
@@ -0,0 +1,40 @@
+import * as assert from "assert";
```
Comment (tylerlevine) [inline modules/utxo-lib/test/integration_local_rpc/generate/outputScripts.ts:1]: `@prettier`

**Review by tylerlevine — COMMENTED** [review]
small nits, feel free to handle in follow up

File: modules/utxo-lib/package.json (hunk 3 above)
Comment (OttoAllmendinger) [reply modules/utxo-lib/package.json:55]: it already is a regular dep (L44)

**Review by tylerlevine — APPROVED** [review]

**Pull request merged**